src/state/slices/opportunitiesSlice/opportunitiesApiSlice.ts

```diff
@@ -151,131 +150,133 @@ export const opportunitiesApi = createApi({
+              )
 
-          if (!resolver) {
-            throw new Error(`resolver for ${defiProvider}::${defiType} not implemented`)
-          }
+              if (!resolver) {
+                console.warn(`resolver for ${defiProvider}::${defiType} not implemented`)
+                return
+              }
 
-          const onInvalidate = (userStakingId: UserStakingId) =>
-            dispatch(opportunities.actions.invalidateUserStakingOpportunity(userStakingId))
+              // TODO: collect and dispatch once to improve perf locally
+              const onInvalidate = (userStakingId: UserStakingId) =>
+                dispatch(opportunities.actions.invalidateUserStakingOpportunity(userStakingId))
 
-          const resolved = await resolver({
-            opportunityIds,
-            defiType,
-            accountId,
-            reduxApi: { dispatch, getState },
-            onInvalidate,
-          })
+              const resolved = await resolver({
+                opportunityIds,
+                defiType,
+                accountId,
+                reduxApi: { dispatch, getState },
+                onInvalidate,
+              })
 
-          if (resolved?.data) {
-            dispatch(opportunities.actions.upsertUserStakingOpportunities(resolved.data))
-          }
+              if (resolved?.data) {
+                // TODO: collect and dispatch once to improve perf locally
+                dispatch(opportunities.actions.upsertUserStakingOpportunities(resolved.data))
+              }
 
-          const byAccountId = {
-            [accountId]: Object.keys(resolved?.data.byId ?? {}).map(
-              userStakingId => deserializeUserStakingId(userStakingId as UserStakingId)[1],
-            ),
-          } as OpportunityDataById
+              const byAccountId = {
+                [accountId]: Object.keys(resolved?.data.byId ?? {}).map(
+                  userStakingId => deserializeUserStakingId(userStakingId as UserStakingId)[1],
+                ),
+              } as OpportunityDataById
 
-          const data = {
-            byAccountId,
-            type: defiType,
-          }
+              const data = {
+                byAccountId,
+                type: defiType,
+              }
 
-          dispatch(opportunities.actions.upsertOpportunityAccounts(data))
+              // TODO: collect and dispatch once to improve perf locally
+              dispatch(opportunities.actions.upsertOpportunityAccounts(data))
+            } catch (e) {
+              const message = e instanceof Error ? e.message : 'Error getting opportunities data'
+              console.error(message)
+            }
+          }),
+        )
 
-          return { data }
-        } catch (e) {
-          const message = e instanceof Error ? e.message : 'Error getting opportunities data'
-          return {
-            error: {
-              error: message,
-              status: 'CUSTOM_ERROR',
-            },
-          }
-        }
```

Comment:
We need to keep the error-handling here - a rejected query is expected within the flow, and will be superseeded by succesful ones

Reply:
Also related to this, `selectOpportunitiesApiQueriesByFilter` will now silently break, meaning we will lose the loading disabled state in savers:

https://github.com/shapeshift/web/blob/32aad5fbf182518bff1a7e97b0daf1956bced388/src/state/slices/opportunitiesSlice/selectors/index.ts#L29-L30

Reply:
🧠 ty for high quality review, saving us from production pain

Reply:
I've updated `selectOpportunitiesApiQueriesByFilter ` to handle the now array args in queries, see 08ca2b13c8312c52fd1d11a6d2f8872da652ba48.

It looks like we aren't actually using a rejected query state for anything, but instead introspecting pending status on the queries as a whole, which should propagate though now. Do we actually need per-query rejected status?

Reply:
Even though we fetch many in a single action, there is still a case of many queries that will be fired for the same endpoint, so introspecting all queries is absolutely valid! And so is using a `queryStatusParam`, we still want to check for the pending status of at least one

Reply:
With regards to the rejected query filter, we want to properly reject RTK queries or there will be bugs waiting to happen. tl;dr is the first succesful query gets cached for given serialized cached args, but rejected ones don't cache. Resolving with `data: null` instead of rejecting is going to rug the follow-up queries

Reply:
Fixed ETH/FOX staking in https://github.com/shapeshift/web/pull/5343/commits/02f75edaadbd9f7eb2f7bed6d7b5cc9e639296eb

Reply:
Proof Gomes is a wizard